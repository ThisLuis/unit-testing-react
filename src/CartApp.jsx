
export const CartApp = () => {
  return (
    <h1>CartApp!!!</h1>
  );
}
