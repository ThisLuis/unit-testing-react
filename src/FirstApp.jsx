import PropTypes from 'prop-types';

export const FirstApp = ( {title, subTitle, name} ) => {
    return (
        <>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <h1>{ title }</h1>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired
}

// Las defaultProps se ejecutan primero que las propTypes
FirstApp.defaultProps = {
    name: 'Luis Nieves!!!',
    subTitle: 'Subtitulo por defecto',
    // title: 'Titulo principal',
}
