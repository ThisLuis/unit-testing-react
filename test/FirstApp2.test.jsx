import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
    
    const title    = 'Hola, soy un test';
    const subTitle = 'Soy un subtitulo';

    test('Debe de hacer match con el snap', () => {
        const { container } = render(<FirstApp title={ title }/>)
        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje Hola, soy un test', () => {
        render(<FirstApp title={ title } />);
        // el objeto screen muestra lo que se a renderizado con la funcion render, en este caso el componente, y podemos acceder a los metodos de la misma manera
        expect( screen.getByText(title)).toBeTruthy()
    });

    test('debe de mostrar el titulo en un h1', () => {
        render(<FirstApp title={ title } />);
        // obtener por role, es basicamente elementos html
        expect( screen.getByRole('heading', { level: 1}).innerHTML).toContain( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        render(
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />
        );

        expect( screen.getAllByText(subTitle).length).toBe(2);
    });
})