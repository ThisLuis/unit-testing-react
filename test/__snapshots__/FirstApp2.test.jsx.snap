// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snap 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy un test
     
  </h1>
  <p>
    Subtitulo por defecto
  </p>
  <p>
    Subtitulo por defecto
  </p>
  <p>
    Luis Nieves!!!
  </p>
</div>
`;
